s3-connector/unittest_s3-connector.py

```diff
@@ -20,62 +20,96 @@ def setUpClass(self):
         self.secret= self.test_cfg['secret']
         self.s3_endpoint=self.test_cfg['s3_endpoint']
         self.region_name=self.test_cfg['region_name']
-        # Intialize connection for S3
+       
         self.s3_connection = s3connector.S3(
             self.test_cfg["key"], 
             self.test_cfg["secret"], 
             self.test_cfg["s3_endpoint"], 
             self.test_cfg["region_name"]
             )
 
-    # Test connection for S3
+    # test connection for s3 resource
     def test_connection(self):
         self.assertTrue(self.s3_connection)
+        
+    #  test connection with incorrect keys
+    def test_conn_with_incorrectkey(self):
+        try:
+          fun = s3connector.S3( 
+            self.test_cfg["incorrect_key"],
+            self.test_cfg["incorrect_secret"],
+            self.test_cfg["s3_endpoint"],
+            self.test_cfg["region_name"]
+            ).list_buckets()
+          for bucket in fun:
+            print(bucket)
+        except ClientError as e:
+            print("Could not connect to s3 resource, check your parameters:",e.response["Error"]["Code"])
+            error = e.response["Error"]["Code"]
+        self.assertEqual(error,"InvalidAccessKeyId")
 
-    # Test list buckets from s3 
+    # test list buckets from s3 resource
     def test_list_bucket(self):
         conn = self.s3_connection.list_buckets()
-        for bucket in conn:
-            print(bucket.name)
-        self.assertTrue(conn)
-        
-    # Test list objects from s3 Bucket
+        for buk in conn:
+            self.assertGreater(len(buk.name), 0)
+           
+  
+    # test non_existing list buckets
+    def test_list_nonexisting_bucket(self):
+        try:
+          conn = self.s3_connection.list_objects(self.test_cfg['bucket_name2'])
+          for b in conn:
+            print(b)
+        except ClientError as e:
+            error = e.response["Error"]["Code"] 
+            print("bucket not found, please check the bucket name")
+            self.assertEquals(error,"AccessDenied")  
+   
+    # test list objects from s3 bucket
     def test_list_object(self):
-        conn = self.s3_connection.list_objects(self.test_cfg['bucket_name'])
-        for obj in conn: 
-            print(obj.key)
-        self.assertTrue(conn)
-
-    # Test bucket versioning
-    def test_check_bucket_versioning(self):
-        conn = self.s3_connection.check_bucket_versioning(self.test_cfg['bucket_name'])
+         conn = self.s3_connection.list_objects(self.test_cfg['bucket_name1'])
+         for obj in conn:
+            self.assertGreater(len(obj.key), 0) 
```

Comment:
need to tighten this test Make it larger than 20 or 15

Reply:
@orionsBeltWest As we are calling list object function , it will show list objects in the bucket more than one. Basicaly we have no idea if bucket has objects or not so, I have mentioned graterthan 1.